docs/testing-tool.rst

```diff
@@ -7,85 +7,118 @@ Use the KSQL testing tool to test a set of KSQL statements. The KSQL testing too
-The test file format is the same as the files KSQL code uses for integration tests. For more details on the
-structure of the test file and all possible settings, see the `README.md <https://github.com/confluentinc/ksql/tree/master/ksql-functional-tests>` in the KSQL repository.
+Output File
+----------
+
+The output file is a JSON file with an array field named "outputs". Similar to the input file, each element in the array is the representation of the expected output messages.
+An expected output message should have a topic, a key, a value and a timestamp. The following is a sample expected output file for the above test:
 
-Running Tests
-*************
+.. code:: json
+     {
+       "outputs": [
+               {"topic": "SRTREAM1", "key": 101, "value": "101"}
+       ]
+     }
 
-Assume we run the previous test file, which is stored in test.json in the home directory.
-The following command shows how to run the test from the terminal:
 
-.. code:: bash
+In the output output files you can have messages with windowed keys. Such messages can be generated by windowed aggretations in KSQL.
+To specify a window for a message you can add a "window" field to the message. A window field has three fields:
+
+- start: the start time for the window.
+- end: the end time for the window.
+- type: the type of the window. A window type can be ``time`` or ``session``.
 
-    ksql-testing-tool ~/test.json
+The following is an example expected output file with records that have a window field:
 
+.. code:: json
+     {
+        "outputs": [
+          {"topic": "S2", "key": 0, "value": "0,0", "timestamp": 0, "window": {"start": 0, "end": 30000, "type": "time"}},
+          {"topic": "S2", "key": 0, "value": "0,5", "timestamp": 10000, "window": {"start": 0, "end": 30000, "type": "time"}},
+          {"topic": "S2", "key": 100, "value": "100,100", "timestamp": 30000, "window": {"start": 30000, "end": 60000, "type": "time"}},
+          {"topic": "S2", "key": 100, "value": "100,100", "timestamp": 45000, "window": {"start": 30000, "end": 60000, "type": "time"}}
+        ]
+     }
+
+Currently, in the input files you can only have record with session window types.
 
-Your output should resemble:
+The testing tool will indicate the success or failure of a test by printing the corresponding message. The following is the result of a successful test:
 
 .. code:: bash
+    ksql-test-runner -s statements.sql -i input.json -o output.json
+     Test passed!
 
-     >>> Running test: ksql-test - my first positive test
-    	 >>> Test ksql-test - my first positive test passed!
-     >>> Running test: ksql-test - my first negative test
-    	 >>> Test ksql-test - my first negative test passed!
-    All tests passed!
+If a test fails, the testing tool will indicate the failure along with the cause. Here is an example of the output for a failing test:
 
+.. code:: bash
+    ksql-test-runner -s statements.sql -i input.json -o output.json
```

Comment:
```suggestion
    ksql-test-runner -s statements_bad.sql -i input_bad.json -o output_bad.json
```
To avoid confusion where someone not reading carefully might expect the above example to fail.